src/components/application_manager/test/commands/hmi/navi_set_video_config_request_test.cc

```diff
@@ -0,0 +1,225 @@
+    ON_CALL(app_mngr_, application(_)).WillByDefault(Return(mock_app_ptr_));
+    ON_CALL(app_mngr_, application_by_hmi_app(_))
+        .WillByDefault(Return(mock_app_ptr_));
+    ON_CALL(app_mngr_, event_dispatcher())
+        .WillByDefault(ReturnRef(mock_event_dispatcher_));
+    ON_CALL(*mock_app_ptr_, hmi_app_id()).WillByDefault(Return(kHmiAppId));
+  }
+
+  MOCK(am::MockHmiInterfaces) mock_hmi_interfaces_;
+  MockAppPtr mock_app_ptr_;
+  MockEventDispatcher mock_event_dispatcher_;
+};
+
+TEST_F(NaviSetVideoConfigRequestTest, OnEvent_SUCCESS) {
+  MessageSharedPtr request_msg = CreateMessage();
+  (*request_msg)[am::strings::msg_params][am::strings::app_id] = kAppId;
+
+  NaviSetVideoConfigRequestPtr command =
+      CreateCommand<NaviSetVideoConfigRequest>(request_msg);
+
+  MessageSharedPtr event_msg = CreateMessage();
+  (*event_msg)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::SUCCESS;
+  Event event(kEventID);
+  event.set_smart_object(*event_msg);
+
+  std::vector<std::string> empty;
+  EXPECT_CALL(
+      app_mngr_,
+      OnStreamingConfigured(
+          kHmiAppId, protocol_handler::ServiceType::kMobileNav, true, empty))
+      .Times(1);
+
+  command->on_event(event);
+}
+
+static bool ValidateList(std::vector<std::string>& expected,
+                         std::vector<std::string>& actual) {
+  if (expected.size() != actual.size()) {
+    return false;
+  }
+  for (unsigned int i = 0; i < expected.size(); i++) {
+    std::string& param = expected[i];
+    unsigned int j;
+    for (j = 0; j < actual.size(); j++) {
+      if (param == actual[j]) {
+        break;
+      }
+    }
+    if (j == actual.size()) {
+      // not found
+      return false;
+    }
+  }
+  return true;
+}
+
+TEST_F(NaviSetVideoConfigRequestTest, OnEvent_FAILURE) {
+  MessageSharedPtr request_msg = CreateMessage();
+  (*request_msg)[am::strings::msg_params][am::strings::app_id] = kAppId;
+  (*request_msg)[am::strings::msg_params][am::strings::config] =
+      smart_objects::SmartObject(smart_objects::SmartType_Array);
+  (*request_msg)[am::strings::msg_params][am::strings::config]
+                [am::strings::protocol] =
+                    hmi_apis::Common_VideoStreamingProtocol::RTP;
+  (*request_msg)[am::strings::msg_params][am::strings::config]
+                [am::strings::codec] =
+                    hmi_apis::Common_VideoStreamingCodec::H265;
+  (*request_msg)[am::strings::msg_params][am::strings::config]
+                [am::strings::height] = 640;
+  (*request_msg)[am::strings::msg_params][am::strings::config]
+                [am::strings::width] = 480;
+
+  NaviSetVideoConfigRequestPtr command =
+      CreateCommand<NaviSetVideoConfigRequest>(request_msg);
+
+  MessageSharedPtr event_msg = CreateMessage();
+  (*event_msg)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::REJECTED;
+
+  (*event_msg)[am::strings::msg_params]["rejectedParams"] =
```

Comment:
Done in https://github.com/smartdevicelink/sdl_core/pull/1700/commits/7b8a976a12c628da2d6bfa8cd3b2f353475d6e11.